pkg/osquery/table/table_util.go

```diff
@@ -4,25 +4,80 @@ import (
 	"io/ioutil"
 	"os"
 	"path/filepath"
+	"runtime"
 
 	"github.com/pkg/errors"
 )
 
-const userDir = "/Users"
+type findFile struct {
+	username string
+}
+
+type FindFileOpt func(*findFile)
+
+func WithUsername(username string) FindFileOpt {
+	return func(ff *findFile) {
+		ff.username = username
+	}
+}
+
+var homedirLocations = map[string][]string{
+	"windows": {"/Users"}, // windows10 uses /Users
+	"darwin":  {"/Users"},
+	"default": {"/home"},
+}
 
 // findFileInUserDirs looks for the existence of a specified path as a
-// subdirectory of users' home directories on macOS
-func findFileInUserDirs(path string) ([]string, error) {
-	userDirs, err := ioutil.ReadDir(userDir)
-	if err != nil {
-		return nil, errors.Wrap(err, "Reading /Users/")
+// subdirectory of users' home directories. It does this by searching
+// likely paths
+func findFileInUserDirs(path string, opts ...FindFileOpt) ([]string, error) {
+	ff := &findFile{}
+
+	for _, opt := range opts {
+		opt(ff)
 	}
-	var res []string
-	for _, dir := range userDirs {
-		fullPath := filepath.Join(userDir, dir.Name(), path)
+
+	var foundDirs []string
+
+	homedirRoots, ok := homedirLocations[runtime.GOOS]
+	if !ok {
+		homedirRoots, ok = homedirLocations["default"]
+		if !ok {
```

Comment:
this seems odd, it should never return !ok, because the homeDirLocations explicitly defines a value for the `"default"` key. Probably not a big deal, although the error message is a bit misleading, it should perhaps reference the fact that the `"default"` key was removed from the homeDirLocations map.

Reply:
This one is me being super defensive. I hate ignore `ok`. Agree on a more clear error